fearless/Common/Services/ChainRegistry/RuntimeProviderPool/RuntimeProvider.swift

```diff
@@ -195,6 +200,27 @@ extension RuntimeProvider: RuntimeProviderProtocol {
 
         resolveRequests()
     }
+
+    func fetchCoderFactoryOperation() -> BaseOperation<RuntimeCoderFactoryProtocol> {
+        ClosureOperation { [weak self] in
+            var fetchedFactory: RuntimeCoderFactoryProtocol?
+
+            let semaphore = DispatchSemaphore(value: 0)
+
+            self?.fetchCoderFactory(runCompletionIn: nil) { [weak semaphore] factory in
```

Comment:
`fetchCoderFactory` stores reference to `semaphore`, but `semaphore` doesn't. I guess `[weak semaphore]` is redundant